docs/logicapps-ai-course/02_build_conversational_agents/07-deploying-agents-to-ms-teams.md

```diff
@@ -0,0 +1,214 @@
+1. Save the file
+
+### Deploy to Azure:
+
+1. **Deploy to Azure App Service**
+   ```bash
+   # Deploy using Azure CLI or Visual Studio publish
+   az webapp deployment source config-zip --resource-group myResourceGroup --name myAppServiceName --src myapp.zip
+   ```
+   To publish in Visual Stuio, right-click on the solution and select **Publish**. In the Publish dialog: Choose Azure as the target. Select the specific deployment target (Windows). Follwing the wizard to deploy the application to Azure App Service. 
+   
+   Once deployed, go to the [Azure portal](https://portal.azure.com) and navigate to your App Service resource. Find and note the **default domain** - this values will be used in the next section.
+
+### Update the Bot Service:
+
+1. **Update Bot Service Configuration**
+   - In the [Azure portal](https://portal.azure.com), navigate to your Bot service.
+   - Go to the **Configuration** page under the **Settings** menu
+   - Update the **Messaging endpoint** with the **default domain** values from the previous set and append the following to the URL: **"/api/messages"**
+   - The URL should have the following structure: `https://your-app-service.azurewebsites.net/api/messages`
+   - Click **Save**
+   - Navigate to the **Channels** page under the **Settings** menu group
+   - Click **Microsoft Teams** channel
+   - Accept terms of service
+   - Click **Apply**
+
+
+## Step 5: Configure Teams Integration
+   
+### Upload Teams App Manifest:
+
+2. **Required Files**:
+   - `manifest.json` (configuration file)
+   - `color.png` (192x192 color icon)
+   - `outline.png` (32x32 outline icon)
+
+3. Update files
+   - Go back to your Visual Studio solution
+   - Open the **manifest.json** file under the **appManifest** folder
+   - Replace **{{BOT_CLIENT_ID}}** with your Bot client ID
+   - Replace **{{BOT_TENANT_ID}}**
+   - Replace **{{AGENT_DISPLAY_NAME_SHORT}}** with **Test Logic App Agent**
+   - Replace **{{AGENT_DISPLAY_NAME_LONG}}** with **Test Logic App Agent**
+   - Save the file
+
+3. **Package Creation**:
+   - Compress all files under **appManifest** into a ZIP archive
+   - Ensure ZIP contains files at root level (not in subfolder)
+
+### Sideload Teams App:
+
+2. **Upload in Teams**
+   - In Teams client: Apps > Manage your apps > Upload a custom app
+   - Select **Upload for [your organization]**
+   - Choose your ZIP file
+
+## Step 6: Testing and Validation
+
+### Test Scenarios:
+A new chat will appear in you chat list with the name of your bot. Open this chat and type **hello** to being your chat session with your Logic Apps agent.
+
+1. **Basic Conversation**
+   - Send simple messages and verify responses
+   - Check authentication flow (OAuth sign-in)
+   - Verify conversation continuity
+
+3. **Error Handling**
+   - Test with invalid inputs
+   - Verify graceful error recovery
+   - Check conversation reset functionality
+
+### Troubleshooting:
```

Comment:
can we expand the troubleshooting section bit more and add types of errors folks would run into and how to solve them?

For example, bot app getting 401 errors means and auth is not configured correctly.

404 means url is not right on the bot service etc